annotation/annotation/tasks/services.py

```diff
@@ -62,6 +62,33 @@
     ValidationSchema,
 )
 
+
+class LRU(OrderedDict):
+    def __init__(self, capacity):
+        self.capacity = capacity
+
+    def __contains__(self, item):
+        if super().__contains__(item):
+            self.move_to_end(item)
+            return True
+        else:
+            return False
+
+    def __getitem__(self, key):
+        if key in self:
+            self.move_to_end(key)
+            return super().__getitem__(key)
+        raise KeyError
+
+    def __setitem__(self, key, value):
+        if key in self:
+            self.move_to_end(key)
+        super().__setitem__(key, value)
```

Comment:
```suggestion
        self[key] = value
```

Reply:
Cannot do that, it creates infinite recursion, since we are calling the method we are overriding.

Reply:
Got it, super()[key] = something would work, but not that clear